### PR TITLE
Add config options for MusicBrainz

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

File: MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace MediaBrowser.Providers.Plugins.MusicBrainz
+{
+    public class PluginConfiguration : BasePluginConfiguration
+    {
+        public bool Enable { get; set; } = false;
+
+        public bool ReplaceArtistName { get; set; } = false;
+
+        public string Server { get; set; } = "https://www.musicbrainz.org";
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs]: Maybe set the default values in the ctor, And no need to set bools to false.

File: MediaBrowser.Providers/Plugins/MusicBrainz/AlbumProvider.cs [hunk 2]
```diff
@@ -74,6 +74,12 @@ public class MusicBrainzAlbumProvider : IRemoteMetadataProvider<MusicAlbum, Albu
         /// <inheritdoc />
         public async Task<IEnumerable<RemoteSearchResult>> GetSearchResults(AlbumInfo searchInfo, CancellationToken cancellationToken)
         {
+            // TODO maybe remove when artist metadata can be disabled
+            if (!Plugin.Instance.Configuration.Enable)
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/AlbumProvider.cs]: These are an ugly hack imo

Comment (dkanada) [reply MediaBrowser.Providers/Plugins/MusicBrainz/AlbumProvider.cs]: I agree but there's no easy alternative at the moment to disable this provider.

File: MediaBrowser.Providers/MediaBrowser.Providers.csproj [hunk 3]
```diff
@@ -24,4 +24,9 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Remove="Plugins\MusicBrainz\Configuration\config.html" />
```
Comment (JustAMan) [inline MediaBrowser.Providers/MediaBrowser.Providers.csproj]: what this does?

File: MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/config.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    <div data-role="page" class="page type-interior pluginConfigurationPage musicBrainzConfigPage" data-require="emby-input,emby-button,emby-checkbox">
+        <div data-role="content">
+            <div class="content-primary">
+                <form class="musicBrainzConfigForm">
+                    <div class="inputContainer">
+                        <input is="emby-input" type="text" id="server" label="Server:" />
+                        <div class="fieldDescription">This can either be a mirror of the official server or a custom server.</div>
+                    </div>
+                    <div class="inputContainer">
+                        <input is="emby-input" type="number" id="rateLimit" pattern="[0-9]*" required min="400" max="10000" label="Rate Limit:" />
```
Comment (JustAMan) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/config.html]: why min 400? how can I turn off rate limiting if I, say, host my own server?

File: MediaBrowser.Providers/Plugins/MusicBrainz/ExternalIds.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    public class MusicBrainzReleaseGroupExternalId : IExternalId
+    {
+        /// <inheritdoc />
+        public string Name => "MusicBrainz Release Group";
+
+        /// <inheritdoc />
+        public string Key => MetadataProviders.MusicBrainzReleaseGroup.ToString();
+
+        /// <inheritdoc />
+        public string UrlFormatString => "https://musicbrainz.org/release-group/{0}";
```
Comment (JustAMan) [inline MediaBrowser.Providers/Plugins/MusicBrainz/ExternalIds.cs]: why this doesn't use server from the config? here and in other similar classes

File: MediaBrowser.Providers/MediaBrowser.Providers.csproj (hunk 3 above)
Comment (dkanada) [reply MediaBrowser.Providers/MediaBrowser.Providers.csproj]: Common style in all plugins to remove the web assets from IDEs and such. The line right below adds it as an embedded resource in the final build.

File: MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs [hunk 6]
```diff
@@ -1,13 +1,41 @@
...
+        public long RateLimit
+        {
+            get
+            {
+                return rateLimit;
+            }
+
+            set
+            {
+                if (value < 2000u && server == Plugin.Instance.DefaultServer)
```
Comment (JustAMan) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs]:
```suggestion
                if (value < Plugin.Instance.DefaultRateLimit && server == Plugin.Instance.DefaultServer)
```

File: MediaBrowser.Providers/Plugins/MusicBrainz/ExternalIds.cs [hunk 7]
```diff
@@ -13,11 +14,10 @@ public class MusicBrainzReleaseGroupExternalId : IExternalId
         public string Key => MetadataProviders.MusicBrainzReleaseGroup.ToString();
 
         /// <inheritdoc />
-        public string UrlFormatString => "https://musicbrainz.org/release-group/{0}";
+        public string UrlFormatString => Plugin.Instance.Configuration.Server + "/release-group/{0}";
 
         /// <inheritdoc />
-        public bool Supports(IHasProviderIds item)
-            => item is Audio || item is MusicAlbum;
+        public bool Supports(IHasProviderIds item) => item is Audio || item is MusicAlbum;
```
Comment (JustAMan) [inline MediaBrowser.Providers/Plugins/MusicBrainz/ExternalIds.cs]: I liked previous (two-line) way more

File: MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs [hunk 8]
```diff
@@ -9,6 +9,18 @@ namespace MediaBrowser.Providers.Plugins.MusicBrainz
...
     {
+        public static Plugin Instance { get; private set; }
+
+        public override Guid Id => new Guid("8c95c4d2-e50c-4fb0-a4f3-6c06ff0f9a1a");
+
+        public override string Name => "MusicBrainz";
+
+        public override string Description => "Get artist and album metadata from any MusicBrainz server.";
+
+        public string DefaultServer = "https://musicbrainz.org";
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs]: Why not make these defaults const?

Comment (dkanada) [reply MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs]: Add the suggestions and I can add them in batch from my phone.

File: MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
+﻿using MediaBrowser.Model.Plugins;
+
+namespace MediaBrowser.Providers.Plugins.MusicBrainz
+{
+    public class PluginConfiguration : BasePluginConfiguration
+    {
+        private string server = Plugin.Instance.DefaultServer;
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs]: Private fields should start with an `_`

File: MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        public static Plugin Instance { get; private set; }
+
+        public override Guid Id => new Guid("8c95c4d2-e50c-4fb0-a4f3-6c06ff0f9a1a");
+
+        public override string Name => "MusicBrainz";
+
+        public override string Description => "Get artist and album metadata from any MusicBrainz server.";
+
+        public string DefaultServer = "https://musicbrainz.org";
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs]:
```suggestion
        public const string DefaultServer = "https://musicbrainz.org";
```

File: MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+        public long DefaultRateLimit = 2000u;
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Plugin.cs]:
```suggestion
        public const long DefaultRateLimit = 2000u;
```

File: MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs [hunk 12]
```diff
@@ -0,0 +1,44 @@
+﻿using MediaBrowser.Model.Plugins;
+
+namespace MediaBrowser.Providers.Plugins.MusicBrainz
+{
+    public class PluginConfiguration : BasePluginConfiguration
+    {
+        private string _server = Plugin.Instance.DefaultServer;
```
Comment (mark-monteiro) [inline MediaBrowser.Providers/Plugins/MusicBrainz/Configuration/PluginConfiguration.cs]: @dkanada DefaultServer is a const, you should access with `Plugin.DefaultServer` not via `Instance`. Same with the references to `DefaultRateLimit`

> 2020-02-29T07:26:39.2721421Z Plugins/MusicBrainz/Configuration/PluginConfiguration.cs(7,34): error CS0176: Member 'Plugin.DefaultServer' cannot be accessed with an instance reference; qualify it with a type name instead [/home/vsts/work/1/s/MediaBrowser.Providers/MediaBrowser.Providers.csproj]

**Pull request merged**